pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -327,52 +385,80 @@ func (be *AbstractEventConnection) processItem(item interface{}, functionLogger
 		be.functionLogger = nil
 		return nil, errors.Wrapf(err, "Can't encode item: %+v", item)
 	}
-	processingResults, ok := <-be.resultChan
 
+	// if eventTimeout is 0, we wait for response without timeout
+	// if it is set, we wait either for response or the timeout
+	if be.connectionManager.GetConfig().eventTimeout == 0 {
+		processingResults, ok := <-be.resultChan
+		return be.postProcessEventRegularFlow(processingResults, ok)
+	} else {
+		return be.waitForResponseWithTimeout()
+	}
+}
+
+func (be *AbstractEventConnection) waitForResponseWithTimeout() (*result.BatchedResults, error) {
```

Comment:
this and the function below it are private, so please move them below the public functions